src/analysis/retail/warlock/demonology/modules/features/T31AmirdrassilDemonology.tsx

```diff
@@ -0,0 +1,130 @@
+import SPELLS from 'common/SPELLS';
+import { formatPercentage, formatNumber } from 'common/format';
+import { TIERS } from 'game/TIERS';
+import { UpArrowIcon, UptimeIcon } from 'interface/icons';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyDebuffEvent, DamageEvent, RemoveDebuffEvent } from 'parser/core/Events';
+import Enemies from 'parser/shared/modules/Enemies';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import Statistic from 'parser/ui/Statistic';
+import UptimeBar from 'parser/ui/UptimeBar';
+
+class T31AmirdrassilDemonology extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  };
+  protected enemies!: Enemies;
+  has4Piece: boolean;
```

Comment:
small note that doesn't need to be changed in this PR: generally, if something isn't intended to be consumed directly from outside the class, making it a `private` value will help ensure that it isn't exposed.